SECURITY.md

```diff
@@ -1,6 +1,63 @@
+
+**Be mindful of risky model formats**. Give preference to run models with `.safetensors` format, and if you need to run load models on the [picle](https://docs.python.org/3/library/pickle.html) format, make sure to use the `weights_only` option to `torch.load`.
+
+Important Note: The trustworthiness of a model is not binary. You must always determine the proper level of caution depending on the specific model and how it matches your use case and risk tolerance.
+
+### Untrusted inputs during training and prediction
+
+Some models accept various input formats (text, images, audio, etc.) and pre-process them by modifying and/or converting the input. Most of this work is done by a variaty of libraries that provide different levels of security. Based on the security history of these libraries, it's considered safe to work with untrusted inputs for PNG, BMP, GIF, WAV, RAW, RAW_PADDED, CSV and PROTO formats. If you process untrusted data in any other format, we you should employ the following recommendations:
+
+* Sandboxing: Isolate the model process. You can follow the same instructions as given on the section [Untrusted models](#untrusted-models).
+* Updates: Keep your model and libraries updated with the latest security patches.
+
+### Data privacy
+
+**Take special security measures if your model if you train models with sensitive data**. Prioritize [sandboxing](https://developers.google.com/code-sandboxing) your models and:
+- Do not feed sensitive data to untrusted model (even if runs in a sandboxed environment)
+- If you consider publishing a model that was partially trained with sensitive data, be aware that data can potentially be recovered from the trained weights (especially if model overfits).
+
+### Untrusted environments or networks
+
+If you can't run your models in a secure and isolated environment or if it must be exposed to an untrusted network, make sure to take the following security precautions:
+* Confirm the hash of any downloaded artifact (e.g. pre-trained model weights) matches a known-good value
+* Encrypt your data if sending it over the network.
+
+### Multi-Tenant environments
+
+If you intend to run multiple models in parallel with shared memory, it is your responsibility to ensure the models do not interact or access each other's data. The primary areas of concern are tenant isolation, resource allocation, model sharing and hardware attacks.
+
+#### Tenant Isolation
+
+You must make sure that models run separately. Since models can run code, it's important to use strong isolation methods to prevent unwanted access to the data from other tenants.
+
+Separating networks is also a big part of isolation. If you keep model network traffic separate, you not only prevent unauthorized access to data or models, but also prevent malicious users or tenants sending graphs to execute under another tenant’s identity.
```

Comment:
I'm not sure what is the specific issue we want to raise or takeaway from this?
Similar to the above, if someone has to run in a multi-tenant environment where they don't trust each other, they need a lot more care than what is suggested here.

Reply:
See https://github.com/pytorch/pytorch/pull/120531#discussion_r1564108945